tests/base/sacado_product_type_01.cc

```diff
@@ -32,8 +32,6 @@ void check()
 {
   AssertThrow (typeid(typename ProductType<T,U>::type) == typeid(CompareType),
                ExcInternalError());
-  AssertThrow (typeid(typename ProductType<T,U>::type) == typeid(T() * U()),
-               ExcInternalError());
```

Comment:
Actually, I think it would still be necessary to have this property. Does this not hold true? Can you say for which types this condition would be violated?